Object test

stelisevil opened the pull request.

stelisevil committed:
first obbject function

stelisevil committed:
Merge branch 'master' into object-test

stelisevil committed:
Finished Object index.js

Couldnt work out the final one. might need a bit of an explanation or help towards the right answer.

matthewtsinontas commented on objects/index.js:
```diff
@@ -76,13 +76,13 @@ export const getObjectValues = (obj) => {
 */
 export const addIsObjectProperty = (obj) => {
-
+  return obj['isObject'] = true; // I cant get this to work so I'm leaving it with what I thought should work
```
You're on the right lines - 

You need to modify the object as you have done - but you are not returning the modified object from this function; you are returning the result from the assignment of a new property to that object

Consider the following if you will...

```js
// here we have a user object
const user = {
  name: 'Matthew',
  age: 24
};
```
This is me modifying the user object
```js
user.isObject = true;
```
The user object now has this new property associated with it
```js
console.log(user.isObject) // true
```
However, this function needs to add the property to the object and then return the new modified object

stelisevil committed:
fixed broken test! then broke it again

so i wrote something that fixed the test, but I just wanted to double check something because I had an incling that all i'd done is create a new object, so i wrote another test to check if the 'name' property was still there and it failed. so I'm committing this for review because I think it'll be something decent to discuss.

matthewtsinontas commented on objects/test.js:
```diff
@@ -78,6 +78,8 @@ describe('Object test', () => {
       expect(newObj).toHaveProperty('isObject');
       expect(newObj.isObject).toEqual(true);
+      expect(newObj).toHaveProperty('firstName');
+      expect(newObj.isObject).toEqual('Matthew');
```
Read the test on line 82, it doesn't quite make sense 😅

matthewtsinontas commented on objects/index.js:
```diff
@@ -84,5 +84,8 @@ export const joinObjects = (objA, objB) => {
+  let newObj = {
+    isObject: true
+  };
+  return newObj;
```
Going through this line by line:
Line 87->89: I am creating a new object with a property isObject with a value of true
Line 90: Return new object

You are creating and returning a new object, you aren't modifying the existing one

stelisevil replied:
yeah thats why i wrote the extra tests. thought i mightve fucked that up!

matthewtsinontas replied:
Have another crack, im about with an eye on github so will be around

stelisevil committed:
Finished final testing on objects

matthewtsinontas commented on objects/index.js:
```diff
@@ -84,8 +84,8 @@ export const joinObjects = (objA, objB) => {
     isObject: true
   };
-  return newObj;
+  return Object.assign(obj, isObj);
```
Nice, this is actually a second way of solving this problem: 

```js
const newObj = obj;
newObj.isObject = true;
return newObj
```

matthewtsinontas replied:
We need to have a chat about whats going on with these functions too, the difference between modifying an existing object and creating a new one

stelisevil committed:
finished and updated object testing
